pkg/storage/disk.go

```diff
@@ -145,10 +146,35 @@ func (s *Module) DiskWrite(name string, image string) error {
 		return fmt.Errorf("image size is bigger than disk")
 	}
 
-	_, err = io.Copy(file, source)
-	if err != nil {
+	// writing the image concurrently to speedup the previous sequential write.
+	// the sequential write is slow because the data source is from the remote server.
+	var (
+		// use errgroup because there is no point in continuing if one of the goroutines failed
+		group               = new(errgroup.Group)
```

Comment:
should be usable on the zero value or with a context if required

Reply:
> should be usable on the zero value

could you elaborate more on this?

> or with a context if required

not possible for now:
- the func doesn't have context
- `io.Copy` itself doesn't have context.

I think make it context aware will need separate ticket

Reply:
> I think make it context aware will need separate ticket

`DiskWrite` handled by `storaged`, called by `provisiond` through `zbus`.
So, plain Go context will not really work here.
But of course we can start with `context` inside `storaged`.

As a side note, i noticed that `DiskWrite` keep going even tho `provisiond` already cancel the deployment/contract.
So, we already have real cancellation issue over zbus.

Reply:
can you please mention the cancellation problem as a known issue on the top of PR to be easily found later on

Reply:
> can you please mention the cancellation problem as a known issue on the top of PR to be easily found later on

Maybe create separate issue because it is not really specific to this PR